datasources/src/main/java/org/opensearch/sql/datasources/storage/OpenSearchDataSourceMetadataStorage.java

```diff
@@ -71,14 +74,21 @@ public class OpenSearchDataSourceMetadataStorage implements DataSourceMetadataSt
    * @param encryptor Encryptor.
    */
   public OpenSearchDataSourceMetadataStorage(
-      Client client, ClusterService clusterService, Encryptor encryptor) {
+      Client client,
+      ClusterService clusterService,
+      Encryptor encryptor,
+      OpenSearchSettings settings) {
     this.client = client;
     this.clusterService = clusterService;
     this.encryptor = encryptor;
+    this.settings = settings;
   }
 
   @Override
   public List<DataSourceMetadata> getDataSourceMetadata() {
+    if (!isEnabled()) {
+      return Collections.emptyList();
```

Comment:
throw exception?

Reply:
Hey Peng, thanks for the comment!

There are a few places in the code which call `getDataSourceMetadata()` and `getDataSourceMetadata(String)` which I felt it would be better to have them noop and continue to work instead of failing with an exception.

In particular here:
1. When performing a SQL query such as `SELECT * FROM test.index` the result should be a 4xx "Data Source test does not exist" -> https://github.com/opensearch-project/sql/blob/main/core/src/main/java/org/opensearch/sql/analysis/Analyzer.java#L153
2. When performing a PPL query such as `show datasources` the result should be a 2xx with an empty list -> https://github.com/opensearch-project/sql/blob/main/core/src/main/java/org/opensearch/sql/planner/physical/datasource/DataSourceTableScan.java#L48

For direct calls to the data source APIs, I agree there should be an exception thrown, which this PR does.